components/admin/lessons/AdminLessonExerciseCard.tsx

```diff
@@ -0,0 +1,167 @@
+import { get, toUpper } from 'lodash'
+import React, { useState } from 'react'
+import { Collapse, Spinner } from 'react-bootstrap'
+import {
+  Exercise,
+  useDeleteExerciseMutation,
+  User,
+  useRemoveExerciseFlagMutation
+} from '../../../graphql'
+import styles from '../../../scss/adminLessonExerciseCard.module.scss'
+import CopyButton from '../../CopyButton'
+
+type HeaderProps = { user: User; exercise: Exercise }
+const Header = ({ user, exercise }: HeaderProps) => {
+  return (
+    <div className={styles.card__header}>
+      <div className={styles.card__badges}>
+        {exercise.flaggedAt && (
+          <span
+            className={`${styles.card__header__badge} ${styles.card__header__badge__flagged}`}
+          >
+            FLAGGED
+          </span>
+        )}
+        <span
+          className={`${styles.card__header__badge} ${styles.card__header__badge__module}`}
+        >
+          {toUpper(exercise.module.name)}
+        </span>
+      </div>
+      <p className={styles.card__header__username}>{user.username}</p>
+      <div className={styles.card__header__contact__container}>
+        <div className={styles.card__header__contact}>
+          <span>{user.email}</span>
+          <CopyButton value={user.email} color={'primary'} />
+        </div>
+        <div className={styles.card__header__contact}>
+          <span>{user.discordUsername}</span>
+          <CopyButton value={user.discordUsername} color={'primary'} />
+        </div>
+      </div>
+    </div>
+  )
+}
+
+type SectionProps = {
+  header: string
+  paragraph: string
+  explanation?: string
+}
+const Section = ({ header, paragraph, explanation }: SectionProps) => {
+  const [show, setShow] = useState(false)
+
+  return (
+    <div className={styles.card__section}>
+      <h3 className={styles.card__section__header}>{header}</h3>
+      <p className={styles.card__section__desc}>{paragraph}</p>
+      {explanation && (
+        <>
+          <button
+            className={styles['card__section--explanation']}
+            onClick={() => setShow(!show)}
```

Comment:
When updating state based on previous value of the state it would be better pass a function the setState to get access to the latest value of the state and prevent hard to find bugs because of race conditions, batching, etc. https://beta.reactjs.org/apis/usestate#updating-state-based-on-the-previous-state

`setShow(prevShow => !prevShow)` something like this for example.